compiler/rustc_codegen_ssa/src/mir/operand.rs

```diff
@@ -84,6 +84,10 @@ impl<'a, 'tcx, V: CodegenObject> OperandRef<'tcx, V> {
                 let llval = bx.scalar_to_backend(x, scalar, bx.immediate_backend_type(layout));
                 OperandValue::Immediate(llval)
             }
+            ConstValue::ZeroSized => {
+                let llval = bx.zst_to_backend(bx.immediate_backend_type(layout));
+                OperandValue::Immediate(llval)
+            }
```

Comment:
If you expand the diff there's a `if layout.is_zst() {` above that early-returns, so this should've been `unreachable!()` without the illegal `zst_to_backend` implementations existing, looks like.
(and this is the only callsite of `zst_to_backend` AFAICT)

Reply:
Though I think I see how this happened, `scalar_to_backend` had a nonsense `Scalar::Int(ScalarInt::ZST) => {` case from many years ago (at least 4? I wasn't able to easily go far back enough in `git blame`).

And this PR just refactored it into a more obvious spot.